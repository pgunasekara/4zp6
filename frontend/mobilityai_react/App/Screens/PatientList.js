import React from 'react';
import {Platform, StyleSheet, Text, View, FlatList, Button} from 'react-native';
import NavigatorMenu from './Navigator';

import ActionButton from 'react-native-action-button';
import Icon from 'react-native-vector-icons/Ionicons';
import PatientListItem from './PatientListItem';

import { GetPatients } from '../Lib/Api';

export default class PatientList extends React.Component {
    static navigationOptions = ({ navigation }) => {
        redirectOptions = [
            {
                title: "Bands",
                handler : () => alert("Not implemented yet")
            },
            {
                title: "Options",
                handler: () => alert("Not implemented yet")
            },
            {
                title:"Sign Out",
                handler: () => navigation.navigate('Home', {})
            }
        ]

        return {
            title: 'Patients',
            headerLeft: null,
            headerRight : <NavigatorMenu options={redirectOptions}/>
        }
    }

    constructor(props){
        super(props);
        this.state = {};
    }

    componentDidMount() {
        GetPatients().then((patientsJson) => {
            console.log(patientsJson);
            this.setState({patients: patientsJson});
        });
    }

    render() {
        const {navigate} = this.props.navigation;
        return (
            <View style={{flex: 1}}>
                <FlatList
                    data = {this.state.patients}
                    renderItem = {({item, index}) => (
                        <PatientListItem
                            navigation = {this.props.navigation}
                            firstName = {item.firstName}
                            lastName = {item.lastName}
                            firstRow = {index == 0}
                            id = {item.id}
                            deviceId = {item.deviceId}
                        />
                    )}
                />
                
                {/* Rest of the app comes ABOVE the action button component !*/}
                <ActionButton buttonColor="rgba(231,76,60,1)"
                    onPress={() => navigate('PatientForm', {}) }
                    degrees={0}>
                    <Icon name="md-create" style={styles.actionButtonIcon} />
                </ActionButton>
                
            </View>
        );
    }
}

const styles = StyleSheet.create({
    actionButtonIcon: {
      fontSize: 20,
      height: 22,
      color: 'white',
    },
  });
