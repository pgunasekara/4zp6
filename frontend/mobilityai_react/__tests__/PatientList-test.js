import React from 'react';
import PatientList from '../App/Screens/PatientList';

import renderer from 'react-test-renderer';

const fakeNavigation = {
    'navigate': function() {}
};

test('renders correctly', () => {
/*    const tree = renderer.create(
        <PatientList
            navigation = {fakeNavigation}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();*/
});