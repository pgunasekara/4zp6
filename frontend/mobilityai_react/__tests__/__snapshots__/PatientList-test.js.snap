// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "bandId": "",
          "firstName": "Joe",
          "key": "1",
          "lastName": "Johnson",
        },
        Object {
          "bandId": "",
          "firstName": "Ruth",
          "key": "2",
          "lastName": "Reynolds",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderBottomColor": "grey",
                "borderBottomWidth": 1,
                "borderTopColor": "grey",
                "borderTopWidth": 1,
                "paddingBottom": 15,
                "paddingLeft": 5,
                "paddingTop": 15,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 25,
                    "height": 50,
                    "opacity": 1,
                    "overflow": "hidden",
                    "width": 50,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    onLoadEnd={[Function]}
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "height": null,
                          "width": null,
                        },
                        Array [
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  />
                  <View
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Array [
                            undefined,
                            undefined,
                          ],
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#BDBDBD",
                            "flex": 1,
                            "justifyContent": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "color": "#ffffff",
                              "textAlign": "center",
                            },
                            Object {
                              "fontSize": 25,
                            },
                            undefined,
                          ]
                        }
                      >
                        JJ
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "textAlign": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 30,
                    }
                  }
                >
                  Joe Johnson
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderBottomColor": "grey",
                "borderBottomWidth": 1,
                "paddingBottom": 15,
                "paddingLeft": 5,
                "paddingTop": 15,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 25,
                    "height": 50,
                    "opacity": 1,
                    "overflow": "hidden",
                    "width": 50,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    onLoadEnd={[Function]}
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "height": null,
                          "width": null,
                        },
                        Array [
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  />
                  <View
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Array [
                            undefined,
                            undefined,
                          ],
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#BDBDBD",
                            "flex": 1,
                            "justifyContent": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "color": "#ffffff",
                              "textAlign": "center",
                            },
                            Object {
                              "fontSize": 25,
                            },
                            undefined,
                          ]
                        }
                      >
                        RR
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "textAlign": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 30,
                    }
                  }
                >
                  Ruth Reynolds
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#f3f3f3",
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "elevation": undefined,
              "justifyContent": "flex-end",
              "zIndex": undefined,
            },
          ],
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "elevation": undefined,
            "justifyContent": "flex-end",
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": undefined,
          }
        }
      />
      <View
        pointerEvents="box-none"
        style={
          Array [
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "elevation": undefined,
                "justifyContent": "flex-end",
                "zIndex": undefined,
              },
            ],
            Object {
              "alignItems": "flex-end",
            },
            Object {
              "paddingVertical": 30,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "marginHorizontal": 30,
                "zIndex": undefined,
              },
              Object {
                "elevation": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.35,
                "shadowRadius": 3,
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgba(231, 76, 60, 1)",
                  "borderRadius": 28,
                  "height": 56,
                  "width": 56,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 28,
                    "height": 56,
                    "justifyContent": "center",
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "rotate": "0deg",
                      },
                    ],
                    "width": 56,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "rgba(255, 255, 255, 1)",
                      "fontSize": 24,
                      "marginTop": -4,
                    }
                  }
                >
                  +
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
